app/boot_control/common.py

```diff
@@ -0,0 +1,550 @@
+        ["/dev/nvme0n1", "/dev/nvme0n1p1", "/dev/nvme0n1p2"...] will be return
+        """
+        cmd = f"-Pp -o NAME {parent_device}"
+        res = list(
+            map(
+                lambda l: l.split("=")[-1].strip('"'),
+                cls._lsblk(cmd).splitlines(),
+            )
+        )
+
+        # the first line is the parent of the dev family
+        if include_parent:
+            return res
+        else:
+            return res[1:]
+
+    @classmethod
+    def get_dev_size(cls, dev: str) -> int:
+        """Return the size of dev by bytes.
+
+        NOTE:
+            -d: print the result from specified dev only
+            -b: print size in bytes
+            -n: no headings
+        """
+        cmd = f"-dbn -o SIZE {dev}"
+        try:
+            return int(cls._lsblk(cmd))
+        except ValueError:
+            raise ValueError(f"failed to get size of {dev}") from None
+
+    @classmethod
+    def mount_rw(cls, target: str, mount_point: Union[Path, str]):
+        """
+        NOTE: pass args = ["--make-private", "--make-unbindable"] to prevent
+        mount events propagation to/from this mount point.
+        """
+        options = ["rw"]
+        args = ["--make-private", "--make-unbindable"]
+        cls._mount(target, str(mount_point), options=options, args=args)
+
+    @classmethod
+    def bind_mount_ro(cls, target: str, mount_point: Union[Path, str]):
+        """
+        NOTE: pass args = ["--make-private", "--make-unbindable"] to prevent
+        mount events propagation to/from this mount point.
+        """
+        options = ["bind", "ro"]
+        args = ["--make-private", "--make-unbindable"]
+        cls._mount(target, str(mount_point), options=options, args=args)
+
+    @classmethod
+    def umount(cls, target: Union[Path, str], *, ignore_error=False):
+        # first try to check whether the target(either a mount point or a dev)
+        # is mounted
+        if not cls.is_target_mounted(target):
+            return
+
+        # if the target is mounted, try to unmount it.
+        try:
+            _cmd = f"umount -f {target}"
+            subprocess_call(_cmd, raise_exception=True)
+        except CalledProcessError as e:
+            _failure_msg = (
+                f"failed to umount {target}: {e.returncode=}, {e.stderr=}, {e.stdout=}"
+            )
+            logger.warning(_failure_msg)
+
+            if not ignore_error:
+                raise _BootControlError(_failure_msg) from None
+
+    @classmethod
+    def mkfs_ext4(cls, dev: str):
+        try:
+            logger.warning(f"format {dev} to ext4...")
+            _cmd = f"mkfs.ext4 -F {dev}"
+            subprocess_call(_cmd, raise_exception=True)
+        except CalledProcessError as e:
+            _failure_msg = f"failed to apply mkfs.ext4 on {dev}: {e!r}"
+            logger.error(_failure_msg)
+            raise MkfsError(_failure_msg)
+
+    @classmethod
+    def mount_refroot(
```

Comment:
Simply speaking, I don't want otaclient to break current running rootfs.
`mount_refroot` mount current rootfs as `ro`, to avoid **ANY** possible write/changes to the current running system.

Reply:
I thought so, but the comment related to this information is nice to have.

Reply:
Sure!

Reply:
updated in [dcf3976](https://github.com/tier4/ota-client/pull/136/commits/dcf397635cc013c450f8a421769757a838560399).